File: app/src/main/java/com/vegashot/vegashot/http/VegasHotHttpClient.java
package com.vegashot.vegashot.http;

import com.squareup.okhttp.OkHttpClient;
import com.squareup.okhttp.Request;
import com.squareup.okhttp.Response;
import com.vegashot.vegashot.model.ClassSchedule;
import com.vegashot.vegashot.scraper.ScraperException;
import com.vegashot.vegashot.scraper.impl.JsoupScraper;

import java.io.IOException;

public class VegasHotHttpClient {

    private static OkHttpClient client = new OkHttpClient();

    private String get(String url) throws IOException {
        Request request = new Request.Builder()
                .url(url)
                .build();

        Response response = client.newCall(request).execute();
        return response.body().string();
    }

    public ClassSchedule getClassSchedule() {
        String response = null;
        try {
            response = get("http://www.vegashot.com/pages/rainbow");
            return new JsoupScraper().scrapeClassSchedule(response);
        } catch (IOException | ScraperException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/vegashot/vegashot/http/VegasHotHttpClient.kt
package com.vegashot.vegashot.http

import com.squareup.okhttp.OkHttpClient
import com.squareup.okhttp.Request
import com.squareup.okhttp.Response
import com.vegashot.vegashot.model.ClassSchedule
import com.vegashot.vegashot.scraper.ScraperException
import com.vegashot.vegashot.scraper.impl.JsoupScraper

import java.io.IOException

public class VegasHotHttpClient {

    throws(IOException::class)
    private fun get(url: String): String {
        val request = Request.Builder().url(url).build()

        val response = client.newCall(request).execute()
        return response.body().string()
    }

    public fun getClassSchedule(): ClassSchedule? {
        try {
            return JsoupScraper().scrapeClassSchedule(get("http://www.vegashot.com/pages/rainbow"))
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        } catch (e: ScraperException) {
            e.printStackTrace()
            return null
        }

    }

    companion object {

        private val client = OkHttpClient()
    }
}
